### ISSUE TITLE
Clicking on button in HUD advances text

### ISSUE THREAD

**Issue by Falcarius** [loner02/vn-canvas#18]
You're probably aware of this one already, but clicking a link:[set]-type button in a non-modal cform will cause the text to advance. If the button is, say, a mute icon in a HUD, this behaviour is obviously unwanted. Is there an easy way around this?

**Comment by loner02** [comment]
Yes, I'm aware of this (and the other issue you reported). I'll get back to you once I get "real" solutions for these. This one is related to click event handling, while the other (the translate snapping) I think is related to transition time.

Btw, thanks very much for reporting.

**Comment by loner02** [comment]
Bug/Issue: When button is clicked with link as "set", stage/script advances or is updated.
Correct behavior: The stage/script should not advance.

Cause/Reason: The handler of button click sets Stage.pause to false.

Analysis: This is a design limitation as will be explained below.
There are 3 recommended commands to put on a button link: "jump", "set" or "macro. Other commands are possible but not recommended and behavior is unknown.
	Case 1- Button link uses "jump". When button is clicked, jump is executed. After jump, the stage/script is automatically updated. This is to make the stage reflect the effect of the jump, or else the user will have to do one extra click before he actually sees the jump.
	Case 2- Button link uses "set". When button is clicked, the variable is set. After setting variable, it is desired that the stage/script should not automatically update. Note, this is opposite behavior as in Case 1.
	Case 3- Button link uses "macro". When button is clicked, a macro/function call is executed. As macro is user defined, there is no way of knowing what the desired behavior should be after execution. Should stage be paused or not?

Fix: For now, the behavior is to update the stage/script, unless the link is a "set".  In scripts/app/vncanvas-cform.js, change the "button" update code:

```
	button: {
	    _init: function (obj, param) {
	        <snipped>
	    },
	    _update: function (obj, elapsed) {
	        <snipped>
	        if (!Stage.activeForm && Stage.mouseClick && obj.inputFocus) {
	            if (obj.link != null) {
	                obj.link[0](obj.link[1]);
	                // Stage.pause = false;			//<-- change this line
	                Stage.pause = (obj.link[0] == set);	//<-- to this
	            }
	            obj.redraw = true;
	        }
	    },
	    _draw: function (obj) {
	        <snipped>
	    }
	},

```

Do the same for "toggle" button.
